doc/adr/0001-record-architecture-decisions.rst

```diff
@@ -0,0 +1,106 @@
+ADR 1: Record architectural decisions
+=====================================
+
+Date: 2022-06-08
+
+Status
+------
+
+Draft
+
+Context
+-------
+
+We are in search for a means to document our architectural and design decisions
+for all of our components.
+In order to do that, there is practice called architectural decision records ("ADR"),
+that we can integrate into our workflow.
+
+This does not replace actual architecture documentation, but provides people who are contributing:
+
+* the means to understand architectural and design decisions that were made
+* a framework for proposing changes to the current architecture
+
+For each decision, it is important to consider the following factors:
+
+* what we have decided to do
+* why we have made this decision
+* what we expect the impact of this decision to be
+* what we have learned in the process
+
+As we're already using `rST <https://docutils.sourceforge.io/rst.html>`_, `Sphinxdoc <https://www.sphinx-doc.org/en/master/>`_ and `readthedocs <https://readthedocs.org/>`_, it would be practical to
+integrate these ADRs as part of our current documentation infrastructure.
+
+Decision
+--------
+
+* We will use ADRs to document, propose and discuss
+  any important or significant architectural and design decisions.
+
+* The ADR format will follow the format described in `Consequences`_ section.
+
+* We will follow the convention of storing those ADRs as Markdown formatted
```

Comment:
I tried to use Markdown, but it didn't work. It would work locally, but when using the nix-build command, it would get the error:
```
> Warning, treated as error:
> /build/source/adr/index.rst:29:toctree contains reference to nonexisting document 'adr/0002-test'

```

Reply:
change `doc/default.nix` line 4 to contain `".md"` also.

Reply:
up to you :shrug: I think if you don't care much about cross-references it's not a big deal